### PR TITLE
fix PathSeparatedPrefix & optimize the prefix ends with / logic.

### PR DESCRIPTION
**Pull request by qicz**
fix & support #1044 

the upstream receive double "/" needs envoy fix , issue https://github.com/envoyproxy/envoy/issues/26024

### CODE REVIEW

File: internal/xds/translator/route.go [hunk 1]
```diff
@@ -96,9 +97,11 @@ func buildXdsRouteMatch(pathMatch *ir.StringMatch, headerMatches []*ir.StringMat
 				Path: *pathMatch.Exact,
 			}
 		} else if pathMatch.Prefix != nil {
-			if *pathMatch.Prefix == "/" {
+			// when prefix end with "/" use RouteMatch_Prefix
```
Comment (arkodg) [inline internal/xds/translator/route.go:100]:
```suggestion
			// when the prefix ends with "/", use RouteMatch_Prefix
```

File: internal/xds/translator/route.go [hunk 2]
```diff
@@ -96,9 +97,11 @@ func buildXdsRouteMatch(pathMatch *ir.StringMatch, headerMatches []*ir.StringMat
...
+			// https://github.com/envoyproxy/gateway/issues/1044
```
Comment (arkodg) [inline internal/xds/translator/route.go:101]:
```suggestion
```

Comment (qicz) [reply internal/xds/translator/route.go]: resolved

**Comment by qicz** [comment]
@arkodg resolved all, pls review again.

**Review by arkodg — APPROVED** [review]
LGTM thanks @qicz !

**Review by Xunzhuo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**